app/models/statement.rb

```diff
@@ -0,0 +1,33 @@
+class Statement < ApplicationRecord
+  belongs_to :cohort
+  belongs_to :lead_provider
+  has_many :statement_items
+
+  validates :output_fee, presence: true
+  validates :month, numericality: { in: 1..12, only_integer: true }
+  validates :year, numericality: { in: 2020..2050, only_integer: true }
+
+  validate :validate_max_statement_items_count
+
+  state_machine :state, initial: :open do
+    state :open
+    state :payable
+    state :paid
```

Comment:
So is it definitely correct to say that every statement ever takes this route? None are cancelled or closed any other way?

Reply:
I don't think there's ever been a cancelled statement. I believe there is always a statement for each month.
Once the statement is `paid` its no longer added to, and shouldn't change.

Reply:
Ok, makes sense - thanks!